compiler/src/dotty/tools/dotc/transform/Memoize.scala

```diff
@@ -122,15 +122,15 @@ class Memoize extends MiniPhase with IdentityDenotTransformer { thisPhase =>
       denot match {
         case fieldDenot: SymDenotation if sym.annotations.nonEmpty =>
           val cpy = fieldDenot.copySymDenotation()
-          cpy.annotations = sym.annotations
+          cpy.annotations = atPhase(typerPhase)(sym.annotationsCarrying(defn.FieldMetaAnnot))
```

Comment:
Annotations on fields that don't have a meta-annotation should end up on the field. It seems that here we only keep annotations on the field that have an explicit `@field` target.